src/shell.html

```diff
@@ -1224,6 +1224,37 @@
       var Module = {
         preRun: [],
         postRun: [],
+        fdBuf: ['', '', ''],
+        printChars: function(fd, ptr, len) {
```

Comment:
Here I suppose we could do the `printChars: function(fd, buffer, ptr, len)` change mentioned to earlier to avoid having to generate garbage? And then inside the body do `buffer = buffer || HEAPU8;` so that a null coming from asm.js/wasm will refer to `HEAPU8`?

Reply:
Modified printChars signature as you suggest, only with the parameter order changed so that ptr (renamed to str_or_offset) is the only mandatory parameter.